homeassistant/components/livisi/climate.py

```diff
@@ -97,39 +67,25 @@ class LivisiClimate(CoordinatorEntity[LivisiDataUpdateCoordinator], ClimateEntit
     _attr_supported_features = ClimateEntityFeature.TARGET_TEMPERATURE
     _attr_target_temperature_high = MAX_TEMPERATURE
     _attr_target_temperature_low = MIN_TEMPERATURE
-    _attr_has_entity_name = True
 
     def __init__(
         self,
         config_entry: ConfigEntry,
         coordinator: LivisiDataUpdateCoordinator,
-        unique_id: str,
-        manufacturer: str,
-        device_type: str,
-        target_temperature_capability: str,
-        temperature_capability: str,
-        humidity_capability: str,
-        room: str,
-        name: str,
+        device: dict[str, Any],
     ) -> None:
         """Initialize the Livisi Climate."""
-        self.config_entry = config_entry
-        self._attr_unique_id = unique_id
-        self._target_temperature_capability = target_temperature_capability
-        self._temperature_capability = temperature_capability
-        self._humidity_capability = humidity_capability
-        self.aio_livisi = coordinator.aiolivisi
-        self._attr_available = False
-        self._attr_name = name
-        self._attr_device_info = DeviceInfo(
-            identifiers={(DOMAIN, unique_id)},
-            manufacturer=manufacturer,
-            model=device_type,
-            name=room,
-            suggested_area=room,
-            via_device=(DOMAIN, config_entry.entry_id),
-        )
-        super().__init__(coordinator)
+        super().__init__(config_entry, coordinator, device)
+
+        self._target_temperature_capability = self.capabilities["RoomSetpoint"]
+        self._temperature_capability = self.capabilities["RoomTemperature"]
+        self._humidity_capability = self.capabilities["RoomHumidity"]
+
+        # For the livisi climate entities, the device should have the room name from
+        # the livisi setup, as each livisi room gets exactly one VRCC device. The entity
+        # name will always be some localized value of "Climate", so the full element name
+        # in homeassistent will be in the form of "Bedroom Climate"
+        self._attr_device_info["name"] = self._attr_device_info["suggested_area"]
```

Comment:
This is confusing.
I think you should instead add an argument (eg. `*, use_room_as_device_name: bool = False`) to the entity `__init__` method and move the comment there.

Then here you can simply use:
`super().__init__(config_entry, coordinator, device, use_room_as_device_name=True)`

Reply:
Great suggestion. Done.